vault/activity_log.go

```diff
@@ -2396,6 +2412,10 @@ func (a *ActivityLog) precomputedQueryWorker(ctx context.Context) error {
 
 	a.logger.Info("finished computing queries", "month", endTime)
 
+	select {
+	case a.precomputedQueryWritten <- struct{}{}:
+	default:
+	}
```

Comment:
`preconputedQueryWritten` does not have a size limit. Is using `select` required?

Reply:
this is done to ensure that it's a non-blocking write, in case there isn't a consumer reading the channel

Reply:
a channel created without a size is an unbuffered channel, so it will always block

Reply:
Ah, Thanks for the explanation. This is very helpful to know!